web/src/app/users/UserEditDialog.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+import { gql, useMutation } from '@apollo/client'
+import Spinner from '../loading/components/Spinner'
+import FormDialog from '../dialogs/FormDialog'
+import { useSessionInfo } from '../util/RequireConfig'
+import { Checkbox, FormControlLabel } from '@material-ui/core'
+import { nonFieldErrors } from '../util/errutil'
+
+const mutation = gql`
+  mutation($input: UpdateUserInput!) {
+    updateUser(input: $input)
+  }
+`
+
+interface UserEditDialogProps {
+  userID: string
+  role: string
+  onClose: () => void
+}
+
+function UserEditDialog(props: UserEditDialogProps): JSX.Element {
+  const { ready: isSessionReady, userID: currentUserID } = useSessionInfo()
+
+  const [state, setState] = React.useState({
+    isAdmin: props.role === 'admin',
+  })
```

Comment:
this can be simplified from an object to
```suggestion
  const [adminChecked, setAdminChecked] = React.useState(isAdmin)
```

would also have to update the `state.isAdmin` references below though, and make sure that `isAdmin` from `useSessionInfo` is called beforehand for scope

Reply:
We would not be able to use `isAdmin` here from `useSessionInfo`. The reason being `isAdmin` checks whether the logged in user is an admin or not, whereas the use case for the variable here is the check the role of the user whose details page is being viewed.

Reply:
Ooooh that makes sense- I was only thinking in the context of the profile page for the permissions of opening the panel. I didn't realize that that could differ when editing someone else's page. That makes sense, ty!